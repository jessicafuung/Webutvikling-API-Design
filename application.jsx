import React from "react";

export function ListMovies() {
    return <div><h1>List movies</h1>
        <ul>
            <li>Movie 1</li>
            <li>Movie 2</li>
        </ul>

    </div>;
}

export function NewMovieForm() {
    return <form>
        <h1>Add new movie</h1>
        <div>
            Title: <input/>
        </div>
        <div>
            Year: <input/>
        </div>
        <div>
            Plot: <textarea/>
        </div>
    </form>
}