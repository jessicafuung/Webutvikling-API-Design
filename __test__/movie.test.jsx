import ReactDOM from "react-dom";
import React from "react";

import {ListMovies, NewMovieForm} from "../application";


describe("movies application", () => {

    it("shows movie list", () => {
        const element = document.createElement("div");
        ReactDOM.render(<ListMovies />, element)

        expect(element.querySelector("h1").innerHTML)
            .toEqual("List movies");
        expect(element.innerHTML).toMatchSnapshot();
    })


    it("show new movie form", () => {
        const element = document.createElement("div");
        ReactDOM.render(<NewMovieForm />, element);
        expect(element.innerHTML).toMatchSnapshot();
    })
})