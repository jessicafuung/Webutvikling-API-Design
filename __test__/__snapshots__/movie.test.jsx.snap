// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movies application show new movie form 1`] = `"<form><h1>Add new movie</h1><div>Title: <input data-testid=\\"title\\" value=\\"\\"></div><div>Year: <input data-testid=\\"year\\" value=\\"\\"></div><div>Plot: <textarea data-testid=\\"plot\\"></textarea></div></form>"`;

exports[`movies application shows movie list 1`] = `"<div><h1>List movies</h1><ul><li>Movie 1</li><li>Movie 2</li></ul></div>"`;
